crates/viewer/re_component_ui/src/text_log_columns.rs

```diff
@@ -0,0 +1,111 @@
+use re_types::blueprint::{components::TextLogColumn, datatypes::TextLogColumnKind};
+use re_ui::{HasDesignTokens as _, UiExt as _};
+use re_viewer_context::{MaybeMutRef, ViewerContext};
+
+pub fn edit_or_view_columns_singleline(
+    _ctx: &ViewerContext<'_>,
+    ui: &mut egui::Ui,
+    columns: &mut MaybeMutRef<'_, Vec<TextLogColumn>>,
+) -> egui::Response {
+    ui.horizontal(|ui| {
+        let mut first = true;
+        for col in columns.iter() {
+            if !*col.visible {
+                continue;
+            }
+
+            if first {
+                first = false;
+            } else {
+                ui.separator();
+            }
+
+            ui.strong(col.kind.name());
+        }
+    })
+    .response
+}
+
+pub fn edit_or_view_columns_multiline(
+    _ctx: &ViewerContext<'_>,
+    ui: &mut egui::Ui,
+    columns: &mut MaybeMutRef<'_, Vec<TextLogColumn>>,
+) -> egui::Response {
+    match columns {
+        MaybeMutRef::Ref(columns) => columns
+            .iter()
+            .filter(|column| column.visible.into())
+            .map(|column| ui.strong(column.kind.name()))
+            .reduce(|a, b| a.union(b))
+            .unwrap_or_else(|| ui.weak("Empty")),
+        MaybeMutRef::MutRef(columns) => {
+            let mut any_edit = false;
+
+            const ITEM_SPACING: f32 = 8.0;
+            let egui::InnerResponse { mut response, .. } = egui::Frame::new()
+                .corner_radius(ui.visuals().menu_corner_radius)
+                .fill(ui.visuals().tokens().text_edit_bg_color)
+                .inner_margin(egui::Margin {
+                    left: ITEM_SPACING as i8,
+                    right: ITEM_SPACING as i8,
+                    top: ITEM_SPACING as i8,
+                    bottom: (ITEM_SPACING * 0.5) as i8,
+                })
+                .show(ui, |ui| {
+                    let text_height = ui
+                        .style()
+                        .text_styles
+                        .get(&egui::TextStyle::Body)
+                        .map(|s| s.size)
+                        .unwrap_or(0.0);
+                    let sz = egui::vec2(ui.max_rect().size().x, ITEM_SPACING + text_height);
+                    let dnd_res = egui_dnd::dnd(ui, "text_log_columns_dnd").show_sized(
+                        columns.iter_mut().enumerate(),
```

Comment:
should leave a comment on why this enumerates - we don't actually use it but it is used for the id (clever, but sneaky!)